scalameta/contrib/shared/src/main/scala/scala/meta/contrib/ScaladocParser.scala

```diff
@@ -43,6 +45,18 @@ object ScaladocParser {
     comment.content.map(parseRec)
   }
 
+  private[this] def generateHeadingParser(level: Int, f: (String) => DocToken): Parser[DocToken] = {
+    val headingSimbols = "=" * level
+    P(
+      // Code block start
+      headingSimbols
+      // Heading description
+        ~ ((AnyChar ~ !"=").rep ~ AnyChar).!.map(f(_))
```

Comment:
Not sure on how to do that. if it is a wronly formatted header, how I can cause the parser fail in the map? 

With an exception?